### DISCUSSION TITLE
Make all sub-images the same height

### DISCUSSION THREAD

**Discussion by dmkaplan2000** [quarto-dev/quarto-cli#7251] [Q&A]
### Description

I have several sub-images that I am combining into a single panel in a Quarto document that I export to PDF. I would like to force the images to have the same height (but different widths), but everything I have tried to change the relative size of each image seems to do nothing. I tried adding `height` arguments to each image and to the overall figure division, both giving the height as a percent and giving it as a number of pixels. I also tried rescaling the images in an image editing program so they all have the same vertical number of pixels. But neither of these had any effect. I could add a transparent square around each image to force them to appear as if they have the same dimensions, but this seems like a heavy-handed solution to the problem. Is there another way to force all images to appear as the same height in the output image?

A conceptual example of the figure I am trying to create is:

````qmd
---
title: "Test images relative size"
format: pdf
---

::: {#fig-images layout-ncol=2}

![Caption 1](image1.png){#fig-no02 height="300"}

![Caption 2](image2.png){#fig-no21 height="300"}

![Caption 3](image3.png){#fig-no27 height="300"}

![Caption 4](image4.png){#fig-no30 height="300"}

Overall caption.
:::

````

**Comment by mcanouil** [comment]
I don't think you can because `width` is set by default which takes precedence over `height`.
So you have to set the width th get the desired display.

Note that this is a Pandoc thing:

```qmd
![Caption 1](https://github.com/mcanouil/awesome-quarto/blob/main/media/media.png?raw=true){#fig-no02 height="100"}
```

```sh
quarto pandoc --from markdown --to latex index.qmd
```

```latex
\begin{figure}
\centering
\includegraphics[width=\textwidth,height=1.04167in]{https://github.com/mcanouil/awesome-quarto/blob/main/media/media.png?raw=true}
\caption{Caption 1}\label{fig-no02}
\end{figure}
```

**Comment by dmkaplan2000** [comment]
OK, thanks for the information. I ended up using GIMP to ember each image in a box of the same dimensions... Not elegant, but it works...

**Reply by mcanouil** [reply]
FYI, the behaviour I described is described in Pandoc documentation: <https://pandoc.org/MANUAL.html#extension-link_attributes>
![image](https://github.com/quarto-dev/quarto-cli/assets/8896044/068e52e4-0132-4018-96c7-4c563fa7e84a)
